contrib/fwupd.spec.in

```diff
@@ -484,6 +510,52 @@ done
 %config(noreplace)%{_sysconfdir}/fwupd/remotes.d/fwupd-tests.conf
 %endif
 
+%files qubes-vm
+%FWUPD_QUBES_DIR/fwupd-download-updates.sh
+%FWUPD_QUBES_DIR/fwupd_usbvm_validate.py
+
+%files qubes-dom0
+%FWUPD_QUBES_DIR/src/fwupd_receive_updates.py
+%FWUPD_QUBES_DIR/src/qubes_fwupdmgr.py
+%FWUPD_QUBES_DIR/src/fwupd-dom0-update
+%FWUPD_QUBES_DIR/src/qubes_fwupd_heads.py
+%FWUPD_QUBES_DIR/src/__init__.py
+%FWUPD_QUBES_DIR/test/fwupd_logs.py
+%FWUPD_QUBES_DIR/test/test_qubes_fwupdmgr.py
+%FWUPD_QUBES_DIR/test/test_qubes_fwupd_heads.py
+%FWUPD_QUBES_DIR/test/__init__.py
+%FWUPD_QUBES_DIR/test/logs/get_devices.log
+%FWUPD_QUBES_DIR/test/logs/get_updates.log
+%FWUPD_QUBES_DIR/test/logs/help.log
+%FWUPD_QUBES_DIR/test/logs/firmware.metainfo.xml
+%FWUPD_QUBES_DIR/test/logs/metainfo_name/firmware.metainfo.xml
+%FWUPD_QUBES_DIR/test/logs/metainfo_version/firmware.metainfo.xml
+
+%post qubes-dom0
+# Replace the python shebangs if fedora is older then 25
+%if 0%{?fedora} <= 25
+sed -i 's/python3/python36/g' %FWUPD_QUBES_DIR/src/qubes_fwupdmgr.py
+sed -i 's/python3/python36/g' %FWUPD_QUBES_DIR/src/fwupd_receive_updates.py
+sed -i 's/python3/python36/g' %FWUPD_QUBES_DIR/src/qubes_fwupd_heads.py
+sed -i 's/python3/python36/g' %FWUPD_QUBES_DIR/test/test_qubes_fwupdmgr.py
+sed -i 's/python3/python36/g' %FWUPD_QUBES_DIR/test/test_qubes_fwupd_heads.py
+sed -i 's/python3/python36/g' %FWUPD_QUBES_DIR/test/fwupd_logs.py
+%endif
+ln -s -f %FWUPD_QUBES_DIR/src/qubes_fwupdmgr.py /bin/qubes-fwupdmgr
+gpg --import %PKI/GPG-KEY-Linux-Vendor-Firmware-Service
+if xl list sys-usb &>/dev/null; then
+    qvm-run --pass-io sys-usb 'gpg --import %PKI/GPG-KEY-Linux-Vendor-Firmware-Service'
+fi
+if xl list sys-firewall &>/dev/null; then
+    qvm-run --pass-io sys-firewall 'gpg --import %PKI/GPG-KEY-Linux-Vendor-Firmware-Service'
+fi
+if xl list sys-whonix &>/dev/null; then
+    qvm-run --pass-io sys-whonix 'rm /home/user/QubesIncoming/dom0/GPG-KEY-Linux-Vendor-Firmware-Service'
+    qvm-copy-to-vm sys-whonix %PKI/GPG-KEY-Linux-Vendor-Firmware-Service
+    qvm-run --pass-io sys-whonix 'gpg --import /home/user/QubesIncoming/dom0/GPG-KEY-Linux-Vendor-Firmware-Service'
```

Comment:
Just the GPG keys? I think you want the PKCS ones too, no?

Reply:
GPG keys are used three times to verify the metadata and firmware update files. The first time, they are used when the files are downloaded, the second time after copying the files from UpdateVM to Dom0, and the third time when Dom0 copies files to sys-usb VM. For now, we are not using the PKCS keys, but if we want to use `jcat-tool` they are available in standard fwupd directory `/etc/pki/fwupd`

Reply:
I deleted these lines. GPG verification has been replaced with jcat verification.